### ISSUE TITLE
add (task) helper to pass in curried tasks

### ISSUE THREAD

**Issue by machty** [machty/ember-concurrency#29]
Presently the `perform` template helper essentially produces a closure action so that you can pass a task (w curried args) into a component and have it be invoked upon `sendAction` or `this.attrs.actionName()`. It seems desirable to be able to pass in a Task with curried args so that there's feature parity w closure actions, and components maintain the ability to stylize themselves according to the isRunning state of the task.

**Comment by Padchi** [comment]
Would be nice, status on this?

**Comment by machty** [comment]
@Padchi still thinking through some ideas about whether this should be task-helper specific or whether in JS it should be easy to create tasks that are curried versions of other tasks.

**Comment by Padchi** [comment]
I don't think this was my problem tho. I just want to pass down a task to a component, perform it from the component and be able to to see if it's running or not. Maybe I'm doing something wrong but isRunning is undefined:

```
// template 
{{search-input on-input=(perform search)}}

// component
<input type="text" value={{query}} oninput={{action on-input value="target.value"}}>
{{if (get on-input 'isRunning') "never true"}}
```
Skickat från min iPhone

**Comment by machty** [comment]
I think this is what you want (though you might want to rename some things):

```
// template
{{search-input on-input=search}}

// component
<input type="text" value={{query}} oninput={{perform on-input value="target.value"}}>
{{if (get on-input 'isRunning') "never true"}}
```

The `perform` helper produces a regular ol function that's compatible with APIs expecting you to pass in closure actions / functions. The way you had it, you were basically doing a lookup of `isRunning` on a function, rather than a task. If you're writing a component that is task-aware, you can just pass in the task directly, and then call `perform` on the task from within the component's template.

**Comment by Padchi** [comment]
Well that work's except that the value that is passed to the task is the whole oninput event: 
`Event {isTrusted: true, type: "input", target: input, currentTarget: input, eventPhase: 2…}`

**Comment by machty** [comment]
@Padchi I believe that's fixed on master but I still need to cut a release :/ will try and get that out soon.

**Comment by machty** [comment]
`(task)` has now been implemented and released in 0.6.0, as is support for `value="..."`; there is an example for value="" at the bottom of http://ember-concurrency.com/#/docs/writing-tasks

**Comment by Luiz-N** [comment]
hey @machty your addon is amazing. An example of passing in a task directly to a component (like you did above) would be great to have in the docs. Also, I'd love to able to define a task in a route then be able to pass that down to child components. Essentially a e-c version of route-action... (route-task) perhaps?

I haven't tried this but could it be possible in the short term to define the tasks in a route then have actions that simply return the raw task. This way I could use route-action closures that simple return a route task?
